src/front/ccrlExtensions.js

```diff
@@ -8,240 +8,397 @@
-            </div>
-            <div class="col-sm info">
-              <p class="small-margin"><small class="info-header">Depth</small></p>
-              <p class="small-margin info-value" id="kibitzer${kibitzerNum}-depth">0</p>
-            </div>
-            <div class="col-sm info">
-              <p class="small-margin"><small class="info-header">Nodes</small></p>
-              <p class="small-margin info-value" id="kibitzer${kibitzerNum}-nodes">0</p>
-            </div>
-            <div class="col-sm info">
-              <p class="small-margin"><small class="info-header">Nps</small></p>
-              <p class="small-margin info-value" id="kibitzer${kibitzerNum}-nps">0</p>
+  // Utility functions
+  const formatCompactNumber = (number) => {
+    if (number < 1000) return number;
+    const units = ['k', 'M', 'B', 'T'];
+    let unitIndex = -1;
+    while (number >= 1000 && unitIndex < units.length - 1) {
+      number /= 1000;
+      unitIndex++;
+    }
+    return `${number.toFixed(2)}${units[unitIndex]}`;
+  };
+
+  const getFullmoveNumberFromFen = (fen) => {
+    const parts = fen.split(" ");
+    return parts.length === 6 ? parseInt(parts[5]) : 1;
+  };
+
+  const convertPVToAlgebraic = (pv, startingFen) => {
+    if (typeof Chess === 'undefined') {
+      console.error("Chess.js not loaded");
+      return pv;
+    }
+    const chess = new Chess(startingFen);
+    const moves = pv.trim().split(/\s+/);
+    let groups = [];
+    let currentMoveNumber = getFullmoveNumberFromFen(startingFen);
+    let isWhiteToMove = chess.turn() === 'w';
+
+    moves.forEach((move, i) => {
+      let moveObj = chess.move(move, { sloppy: true });
+      if (!moveObj) return;
+
+      if (isWhiteToMove) {
+        groups.push({
+          moveNumber: currentMoveNumber,
+          white: moveObj.san,
+          black: null
+        });
+      } else {
+        if (groups.length > 0) {
+          groups[groups.length - 1].black = moveObj.san;
+        } else {
+          groups.push({
+            moveNumber: currentMoveNumber,
+            white: "...",  // <-- Place "..." after move number when Black moves first
+            black: moveObj.san
+          });
+        }
+        currentMoveNumber++;
+      }
+
+      isWhiteToMove = chess.turn() === 'w';
+    });
+
+    return groups.map(group => {
+      let moveNumberHtml = `<span style="color: #f5c276; font-weight: bold;">${group.moveNumber}.</span>`;
+      let whiteMove = group.white ? `<span style="color: #ffffff;">${group.white}</span>` : '';
+      let blackMove = group.black ? `<span style="color: #ffffff;">${group.black}</span>` : '';
+
+      return `${moveNumberHtml} ${whiteMove} ${blackMove}`.trim();
+    }).join(" ");
+  };
+
+  // A helper function to parse the score string into a numeric value and display string
+  const parseScore = (scoreText) => {
+    scoreText = scoreText.trim();
+    console.log(scoreText)
```

Comment:
Let's maybe comment this bit.

Reply:
Solved by a379391ab1805389ceb32af68ced4610dab72d59